app/src/ui/stashing/stash-diff-viewer.tsx

```diff
@@ -0,0 +1,126 @@
+/* tslint:disable:button-group-order */
+
+import * as React from 'react'
+import { IStashEntry, StashedChangesLoadStates } from '../../models/stash-entry'
+import { FileList } from '../history/file-list'
+import { Dispatcher } from '../dispatcher'
+import { FileChange, CommittedFileChange } from '../../models/status'
+import { Repository } from '../../models/repository'
+import { openFile } from '../lib/open-file'
+import { join } from 'path'
+import { Diff } from '../diff'
+import { IDiff, ImageDiffType } from '../../models/diff'
+import { Resizable } from '../resizable'
+import { Button } from '../lib/button'
+import { ButtonGroup } from '../lib/button-group'
+
+interface IStashDiffViewerProps {
+  /** The stash in question. */
+  readonly stashEntry: IStashEntry
+  /** Currently selected file in the list */
+  readonly selectedStashedFile: CommittedFileChange | null
+  /** Diff to be displayed */
+  readonly stashedFileDiff: IDiff | null
+  readonly imageDiffType: ImageDiffType
+  /** width to use for the files list pane */
+  readonly fileListWidth: number
+  readonly externalEditorLabel?: string
+  readonly onOpenInExternalEditor: (path: string) => void
+  readonly repository: Repository
+  readonly dispatcher: Dispatcher
+}
+
+/**
+ * Component to display a selected stash's file list and diffs
+ *
+ * _(Like viewing a selected commit in history but for a stash)_
+ */
+export class StashDiffViewer extends React.PureComponent<
+  IStashDiffViewerProps
+> {
+  private onSelectedFileChanged = (file: FileChange) =>
+    this.props.dispatcher.changeStashedFileSelection(
+      this.props.repository,
+      file as CommittedFileChange
```

Comment:
Do type check to ensure `file` is of type `CommittedFileChange`

Reply:
Tracked in https://github.com/desktop/desktop/issues/7296